chartfx-chart/src/main/java/de/gsi/chart/legend/spi/DefaultLegend.java

```diff
@@ -106,9 +107,17 @@ protected double computePrefWidth(final double forHeight) {
 
     public LegendItem getNewLegendItem(final Renderer renderer, final DataSet series, final int seriesIndex) {
         final Canvas symbol = renderer.drawLegendSymbol(series, seriesIndex, SYMBOL_WIDTH, SYMBOL_HEIGHT);
-        return new LegendItem(series.getName(), symbol);
+        var item = new LegendItem(series.getName(), symbol);
+        if(renderer.supportsVisibility()){
+            item.setOnMouseClicked(event-> series.setVisible(!series.isVisible()));
+            item.pseudoClassStateChanged(disabledClass, !series.isVisible());
+            series.visibleProperty().addListener((obs, old, newValue) -> item.pseudoClassStateChanged(disabledClass, !newValue));
```

Comment:
This would probably need to changeed to the non-UI event listener, something like:
```java
        series.addListener(evt -> {
            // possible with a guard to listen only to meta event changes and when the state actually changed
            item.pseudoClassStateChanged(disabledClass, !series.isVisible());        
        });
```